six/common/sixstrings.c

```diff
@@ -44,13 +44,18 @@ PyObject *from_json(const char *data) {
         goto done;
     }
 
-    char module_name[] = "json";
+    // JSON return unicode strings under Python2, even if this is a better
+    // behavior it would be a breaking change compare to the previous version
+    // of the Agent not using this lib to embed Python. Yaml returns bytes
+    // under Python2 and Unicode under Python3. Since JSON is a subset of Yaml
+    // we use 'yaml' here.
+    char module_name[] = "yaml";
```

Comment:
JSON is indeed a subset of YAML 1.2. However I am not 100% sure if pyyaml implements 1.2 guidelines:
https://github.com/yaml/pyyaml/issues/116

1.2 was mainly about dealing with some corner case that made JSON "kind of" a subset of YAML. diff are highlighted here: https://github.com/yaml/yaml/wiki/YAML-1.2-Changelog. Some of them might affect us. WDYT ?

Reply:
We completely control the data between the agent and six. It's serialized by the agent and loaded by `six` so we know that escaped `/` would not happened (which is the only one we care about).

But you raise a good point, since we control the whole chain why not use YAML everywhere !

I updated my PR to move entirely to YAML.

Tell me what you think.